### ISSUE TITLE
KF 1.0: KFP UI produces ugly results when not on GKE

### ISSUE THREAD

**Issue by elikatsis** [kubeflow/kubeflow#4829]
/kind bug

**What steps did you take and what happened:**
I deployed KF 1.0 on-prem

KFP UI has a `gkeMetadata` component [since 0.1.39, KF 1.0 ships with KFP 0.2.0] which adds GKE cluster-related information to KFP UI.
It queries `/pipeline/system/cluster-name` and `/pipeline/system/project-id`. The frontend listens to those paths and serves them by making the following queries:
```typescript
const response = await fetch(
  'http://metadata/computeMetadata/v1/instance/attributes/cluster-name',
  { headers: { 'Metadata-Flavor': 'Google' } },
);
```
and `.../project-id` correspondingly.

But they return 404 when not on GKE:
![image](https://user-images.githubusercontent.com/14970053/75911379-26d75280-5e58-11ea-89f9-4b962f99512d.png)

This is fixed for KFP >= 0.2.5. GKE Metadata can be disabled via an environment variable.

Should we bump KFP deployment to version 0.2.5 [and also disable this feature by default in non-gke deployments in manifests] ?

**What did you expect to happen:**
KFP not showing any ugly error

**Comment by Bobgy** [comment]
This also looks like a bug in pipeline UI, when the endpoint returns 404, it shouldn't use its content.

**Comment by Bobgy** [comment]
Sorry for the confusion, I just wrote a fix for this: https://github.com/kubeflow/pipelines/pull/3218.
The fix will be in the next release after 0.2.5, so if after 0.2.5 you won't necessarily need to update manifest to disable the endpoint, but I still recommend doing so for saving network requests.

**Comment by Bobgy** [comment]
With https://github.com/kubeflow/manifests/issues/984 done, I think we can close this.
/close
